test/functionality/dataflow/processing-of-elements/expression-lists/read-tests.ts

```diff
@@ -43,7 +42,7 @@ describe('Lists with variable references', withShell(shell => {
 		const sameGraph = (id1: NodeId, id2: NodeId, definedAt: NodeId) =>
 			emptyGraph()
 				.defineVariable(id1, 'x')
-				.defineVariable(id2, 'x', LocalScope, { environment: define({ nodeId: id1, name: 'x', scope: LocalScope, kind: 'variable', definedAt, used: 'always' }, LocalScope, initializeCleanEnvironments()) })
+				.defineVariable(id2, 'x', LocalScope, { environment: clearEnvironment().define(variable('x', definedAt, id1)) })
```

Comment:
I really really like the idea of the builder and how it plays out. But i have a problem with the naming(s) - so please give me your opinion on that:

1. I think `defaultEnvironment` is the "better" name as (especially with dataflow v2) it already contains R's defaults bindings - furthermore, clear suggests that we empty an existing environment.
2. we now have `.defineVariable` for the graph builder and `.define(variable(...))` for the environments. i think this should be more "synchronous", .e.g with `defaultEnv().defineVariable(...)` or even just `defaultEnv().variable()` with definitions being mostly implicit for the environment. 

Besides: what about a way to specify an environment from an object? 


`env().from({ x: [{ id: id1, at: definedAt}} })` 

or 

`env().add('x', definedAt, id1)` which infers the `variable` from the arguments? 

I am not too sure which is best - feel free to make a proposal :)